src/services/api/scheduling.ts

```diff
@@ -52,7 +51,79 @@ function generateSchedulingConfiguration({
     ),
   }
   console.log('Attempting to generate scheduling configuration', url, body)
-  return postRequest<{ bookingUrl: string }>(url, body).catch((error) => {
+  return postRequest<{ bookingUrl: string }>(url, body)
+}
+
+async function startNylasBooking({
+  formSubmissionResult,
+  schedulingSubmissionEvent,
+  schedulingReceiptUrl,
+  schedulingCancelUrl,
+}: {
+  formSubmissionResult: FormSubmissionResult
+  schedulingSubmissionEvent: SubmissionEventTypes.NylasSubmissionEvent
+  schedulingReceiptUrl: string
+  schedulingCancelUrl: string
+}): Promise<{
+  bookingUrl: string
+}> {
+  const url = `${tenants.current.apiOrigin}/nylas/start-booking`
+  const body = {
+    formId: formSubmissionResult.definition.id,
+    nylasConfigurationId:
+      schedulingSubmissionEvent.configuration.nylasConfigurationId,
+    submissionId: formSubmissionResult.submissionId,
+    schedulingReceiptUrl,
+    schedulingCancelUrl,
+    name: getBookingQuerystringValue(
+      schedulingSubmissionEvent.configuration.nameElementId,
+      formSubmissionResult,
+    ),
+    email: getBookingQuerystringValue(
+      schedulingSubmissionEvent.configuration.emailElementId,
+      formSubmissionResult,
+    ),
+  }
+  console.log('Attempting to generate scheduling configuration', url, body)
+
+  const { queryString } = await postRequest<{ queryString: string }>(url, body)
+
+  return { bookingUrl: `/forms/:formId/scheduling-form?${queryString}` }
```

Comment:
This URL should be defined in the renderer, not apps. As we dont want apps to assume that the renderer has routes structured in a certain way.

Reply:
Ah yes.